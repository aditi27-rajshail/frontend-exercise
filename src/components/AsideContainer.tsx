/**
 * A container component for displaying aside content.
 * @param {AsideContainerData} asideContainerData - Data for the container, including title and subText.
 * @returns {React.Element} - The rendered aside container component.
 */

import { AsideContainerData } from "../types";

interface AsideContainerDataProps {
  asideContainerData: AsideContainerData;
}

const AsideContainer = ({ asideContainerData }: AsideContainerDataProps) => {
  return (
    <div
      data-testid="aside-container"
      className="aside-container flex flex-col items-start w-full xl:w-[189px] p-[16px] mb-[16px]">
      <p className="aside-container-title">{asideContainerData.title}</p>
      <p className="aside-container-sub-text ">{asideContainerData.subText}</p>
    </div>
  );
};

export default AsideContainer;
