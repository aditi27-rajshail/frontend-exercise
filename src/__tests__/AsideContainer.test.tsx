import { render, screen } from "@testing-library/react";
import { describe, it, expect, beforeEach } from "vitest";
import AsideContainer from "../components/AsideContainer";
import { asideContainerData } from "../constants";

describe("AsideContainer Component", () => {
  beforeEach(() => {
    render(<AsideContainer asideContainerData={asideContainerData[0]} />);
  });

  it("renders the aside container title", () => {
    expect(screen.getByText(asideContainerData[0].title).textContent).toBe(
      asideContainerData[0].title
    );
  });

  it("renders the aside container subText", () => {
    expect(screen.getByText(asideContainerData[0].subText).textContent).toBe(
      asideContainerData[0].subText
    );
  });

  it("matches the snapshot", () => {
    const { container } = render(
      <AsideContainer asideContainerData={asideContainerData[0]} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
