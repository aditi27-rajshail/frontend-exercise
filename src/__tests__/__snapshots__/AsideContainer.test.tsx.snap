// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AsideContainer Component > matches the snapshot 1`] = `
<div
  class="aside-container flex flex-col items-start w-full xl:w-[189px] p-[16px] mb-[16px]"
  data-testid="aside-container"
>
  <p
    class="aside-container-title"
  >
    年度初めの残高入力
  </p>
  <p
    class="aside-container-sub-text "
  >
    過去の計算結果を引き継いで計算する場合はこちらをクリックしてください。
  </p>
</div>
`;
